test/pubsub.spec.js

```diff
@@ -6,386 +6,194 @@ chai.use(require('dirty-chai'))
       const message = {
-        from: psA.peerId.id,
+        from: peerInfo.id.id,
         data: 'hello',
         seqno: randomSeqno(),
         topicIDs: ['test-topic']
       }
 
-      sinon.stub(psA, 'strictSigning').value(false)
+      sinon.stub(pubsub, 'strictSigning').value(false)
 
-      psA._buildMessage(message, (err, signedMessage) => {
-        expect(err).to.not.exist()
+      const signedMessage = await pubsub._buildMessage(message)
+      const verified = await pubsub.validate(signedMessage)
 
-        psA.validate(signedMessage, (err, verified) => {
-          expect(verified).to.eql(true)
-          done(err)
-        })
-      })
+      expect(verified).to.eql(true)
     })
 
-    it('validate with strict signing requires a signature', (done) => {
+    it('validate with strict signing requires a signature', async () => {
       const message = {
-        from: psA.peerId.id,
+        from: peerInfo.id.id,
         data: 'hello',
         seqno: randomSeqno(),
         topicIDs: ['test-topic']
       }
 
-      psA.validate(message, (err, verified) => {
-        expect(verified).to.eql(false)
-        done(err)
-      })
-    })
-  })
-
-  describe('dial the pubsub protocol on mount', () => {
-    let nodeA
-    let nodeB
-    let psA
-    let psB
-
-    before((done) => {
-      series([
-        (cb) => createNode(cb),
-        (cb) => createNode(cb)
-      ], (cb, nodes) => {
-        nodeA = nodes[0]
-        nodeB = nodes[1]
-        nodeA.dial(nodeB.peerInfo, () => setTimeout(done, 1000))
-      })
-    })
-
-    after((done) => {
-      parallel([
-        (cb) => nodeA.stop(cb),
-        (cb) => nodeB.stop(cb)
-      ], done)
-    })
-
-    it('dial on pubsub on mount', (done) => {
-      psA = new PubsubImplementation(nodeA)
-      psB = new PubsubImplementation(nodeB)
-
-      parallel([
-        (cb) => psA.start(cb),
-        (cb) => psB.start(cb)
-      ], next)
-
-      function next () {
-        expect(psA.peers.size).to.equal(1)
-        expect(psB.peers.size).to.equal(1)
-        done()
-      }
-    })
+      const verified = await pubsub.validate(message)
 
-    it('stop both pubsubs', (done) => {
-      parallel([
-        (cb) => psA.stop(cb),
-        (cb) => psB.stop(cb)
-      ], done)
+      expect(verified).to.eql(false)
     })
   })
 
-  describe('prevent concurrent dials', () => {
-    let sandbox
-    let nodeA
-    let nodeB
-    let psA
-    let psB
-
-    before((done) => {
-      sandbox = chai.spy.sandbox()
-
-      series([
-        (cb) => createNode(cb),
-        (cb) => createNode(cb)
-      ], (err, nodes) => {
-        if (err) return done(err)
-
-        nodeA = nodes[0]
-        nodeB = nodes[1]
-
-        // Put node B in node A's peer book
-        nodeA.peerBook.put(nodeB.peerInfo)
-
-        psA = new PubsubImplementation(nodeA)
-        psB = new PubsubImplementation(nodeB)
-
-        psB.start(done)
-      })
+  describe('should be able to register two nodes', () => {
```

Comment:
What's the point of this test? Right now it's just validating that a mock is behaving appropriately. This seems like it should just be a test scenario for the registrar itself and not in pubsub.

Reply:
it is basically to guarantee that the `peers` Map is properly updated on connect and disconnect